Engine/DataFeeds/LiveTradingDataFeed.cs

```diff
@@ -498,21 +498,34 @@ protected Subscription CreateSubscription(SubscriptionRequest request)
                             });
                             enumerator = quoteBarAggregator;
                             break;
+
                         case TickType.Trade:
                         default:
                             var tradeBarAggregator = new TradeBarBuilderEnumerator(request.Configuration.Increment, request.Security.Exchange.TimeZone, _timeProvider);
+                            var auxDataEnumerator = new EnqueueableEnumerator<BaseData>();
+
                             _exchange.AddDataHandler(request.Configuration.Symbol, data =>
                             {
-                                var tick = data as Tick;
-
-                                if (tick.TickType == TickType.Trade)
+                                if (data.DataType == MarketDataType.Auxiliary)
                                 {
-                                    tradeBarAggregator.ProcessData(tick);
-                                    if (SubscriptionShouldUpdateRealTimePrice(subscription, timeZoneOffsetProvider)) subscription.RealtimePrice = data.Value;
+                                    auxDataEnumerator.Enqueue(data);
+                                }
+                                else
+                                {
+                                    var tick = data as Tick;
+                                    if (tick.TickType == TickType.Trade)
+                                    {
+                                        tradeBarAggregator.ProcessData(tick);
+                                        if (SubscriptionShouldUpdateRealTimePrice(subscription, timeZoneOffsetProvider)) subscription.RealtimePrice = data.Value;
+                                    }
                                 }
                             });
-                            enumerator = tradeBarAggregator;
+
+                            enumerator = request.Configuration.SecurityType == SecurityType.Equity
+                                ? (IEnumerator<BaseData>) new LiveBaseDataSynchronizingEnumerator(_frontierTimeProvider, request.Security.Exchange.TimeZone, auxDataEnumerator, tradeBarAggregator)
```

Comment:
Is there any harm in using the synchronizing enumerator for all security types?

Reply:
Don't we only have aux data with equities ? Even if the performance hit of the extra enumerator is minimal, I personally prefer to add it only where needed.

Reply:
sounds good -- just where possible I try not to encode how things 'currently are' in multiple places if not required, because in the (currently unlikely) event that things change, it's just more places that introduce bugs later.